freedomofpress/securedrop
# Slims down upgrade testing logic

## Status

Ready for review

## Description of Changes

Resolves a few issues:

* Closes #5512 
* Closes #5883 
* Closes #3208 

Changes proposed in this pull request:

Makes a few changes to refresh the upgrade testing logic for
compatibility with Focal VMs. Specifically:

  * Prod VMs are now configured via Molecule
  * Removes all use of custom-built Vagrant boxes
  * Clarifies upgrade steps (see also related docs PR)

Docs will be updated separately.

## Testing
The docs PR at https://github.com/freedomofpress/securedrop-docs/pull/227 should be reviewed in tandem with this PR. Check those docs and make sure the procedures there work for the use case of SD pre-release QA. At a high-level, those steps are:

0. Make sure you're using libvirt-based VMs (upgrade scenario does not support Qubes env)
1. `molecule create -s libvirt-prod-focal`
2. Boot up admin workstation and install against those prod VMs with `./securedrop-admin install`
3. `make build-debs` on host (ok to run this in parallel with step 2 to save time)
4. `make upgrade-start` on host, to set up local apt repo
5. Back in admin workstation, run the playbook `securedrop-apt-local.yml` (make sure to source the admin venv first, see docs)
6. Confirm you can upgrade packages inside the VMs



## Deployment
Dev-only.
